_includes/v23.1/performance/reduce-contention.md

```diff
@@ -0,0 +1,13 @@
+- Limit the number of affected rows by following [optimizing queries](apply-statement-performance-rules.html) (e.g., avoiding full scans, creating secondary indexes, etc.). Not only will transactions run faster, lock fewer rows, and hold locks for a shorter duration, but the chances of [read invalidation](architecture/transaction-layer.html#read-refreshing) when the transaction's [timestamp is pushed](architecture/transaction-layer.html#timestamp-cache), due to a conflicting write, are decreased because of a smaller read set (i.e., a smaller number of rows read).
+
+- Break down larger transactions (e.g., [bulk deletes](bulk-delete-data.html)) into smaller ones to have transactions hold locks for a shorter duration. For example, use [common table expressions](common-table-expressions.html) to group multiple clauses together in a single SQL statement. This will also decrease the likelihood of [pushed timestamps](architecture/transaction-layer.html#timestamp-cache). For instance, as the size of writes (number of rows written) decreases, the chances of the transaction's timestamp getting bumped by concurrent reads decreases.
+
+- Use [`SELECT FOR UPDATE`](select-for-update.html) to aggressively lock rows that will later be updated in the transaction. Updates must operate on the most recent version of a row, so a concurrent write to the row will cause a retry error ([`RETRY_WRITE_TOO_OLD`](transaction-retry-error-reference.html#retry_write_too_old)). Locking early in the transaction forces concurrent writers to block until the transaction is finished, which prevents the retry error. Note that this locks the rows for the duration of the transaction; whether this is tenable will depend on your workload. For more information, see [When and why to use `SELECT FOR UPDATE` in CockroachDB](https://www.cockroachlabs.com/blog/when-and-why-to-use-select-for-update-in-cockroachdb/).
+
+- Use historical reads ([`SELECT ... AS OF SYSTEM TIME`](as-of-system-time.html)), preferably [bounded staleness reads](follower-reads.html#when-to-use-bounded-staleness-reads) or [exact staleness with follower reads](follower-reads.html#run-queries-that-use-exact-staleness-follower-reads) when possible to reduce conflicts with other writes. This reduces the likelihood of [`RETRY_SERIALIZABLE`](transaction-retry-error-reference.html#retry_serializable) errors as fewer writes will happen at the historical timestamp. More specifically, writes' timestamps are less likely to be pushed by historical reads as they would [when the read has a higher priority level](architecture/transaction-layer.html#transaction-conflicts).
```

Comment:
This may be an entirely fine way to say this, but it may be worth mentioning - if your AOST value is below the closed timestamp, the read can't possibly be invalidated.